core/src/main/java/org/renjin/primitives/combine/LazyBuilder.java

```diff
@@ -63,11 +63,11 @@ private Vector composeNames(String prefix, Vector vector) {
     Vector names = vector.getNames();
     int numElements = vector.length();
 
-    if(!Strings.isNullOrEmpty(prefix) || names != null) {
+    if(!Strings.isNullOrEmpty(prefix) || (names !=  Null.INSTANCE && names != null)) {
       hasNames = true;
     }
 
-    if(Strings.isNullOrEmpty(prefix) && names == null) {
+    if(Strings.isNullOrEmpty(prefix) && (names ==  Null.INSTANCE || names == null)) {
```

Comment:
Sure - done.  Did not add the @NonNull check as it did not seem to be on the classpath ..